### ISSUE TITLE
[error] method:/xxxxxx.antenna.AntennaServer/PushVehicleStatus, proto.Unmarshal:proto: cannot parse invalid wire-format data

### ISSUE THREAD

**Issue by fireblackman** [vearne/grpcreplay#34]
> 2024/08/03 14:04:55 [info] create Http2Conn, MaxDynamicTableSize:4096 2024/08/03 14:04:55 [error] method:/xxxxxx.antenna.AntennaServer/PushVehicleStatus, proto.Unmarshal:proto: cannot parse invalid wire-format data 2024/08/03 14:04:55 [error] get TCP pkg:127.0.0.1:53322 -> 127.0.0.1:6789, seq:2950422669, tcp flags:[PSH ACK] 1 4e71d4fb-515e-11ef-8d38-04421a24b909 1722665095663515270 {"headers":{":authority":"127.0.0.1:6789",":method":"POST",":path":"/xxxxxx.antenna.AntennaServer/PushVehicleStatus",":scheme":"http","content-type":"application/grpc","grpc-accept-encoding":"gzip","grpc-timeout":"3997683u","te":"trailers","user-agent":"grpc-java-netty/1.38.0"},"method":"/xxxxxx.antenna.AntennaServer/PushVehicleStatus","request":"{}"}
> 
> {} 2024/08/03 14:05:00 [info] create Http2Conn, MaxDynamicTableSize:4096
> 
> Some interfaces failed to parse the body, causing forwarding to another server to fail!

2024/08/03 14:04:55 [error] method:/xxxxxx.antenna.AntennaServer/PushVehicleStatus, proto.Unmarshal:proto: cannot parse invalid wire-format data
From the log, it can be seen that the request body deserialization failed. If possible, please provide the definition of the input parameter of: /xxxxxx.antenna.AntennaServer/PushVehicleStatus so that I can further analyze it. @fireblackman

_Originally posted by @vearne in https://github.com/vearne/grpcreplay/issues/26#issuecomment-2266606502_

**Comment by fireblackman** [comment]
The problem remains.

**Comment by vearne** [comment]
> The problem remains, last issue has body parameter.


Please provide a proto file. @fireblackman

**Comment by fireblackman** [comment]
2024/08/05 20:14:19 [info] m: <nil>, ok: false
2024/08/05 20:14:19 [info] parseSymbol, svc:fabupilot.antenna.AntennaServer, method:PushVehicleStatus
2024/08/05 20:14:19 [info] pfd: MessageDescriptor{Syntax: proto2, FullName: fabupilot.antenna.PushVehicleStatusRequest, Fields: [{Name: status, Number: 1, Cardinality: optional, Kind: message, JSONName: "status", HasPresence: true, Message: fabupilot.antenna.VehicleStatus}]}
2024/08/05 20:14:19 [info] pfmd: MessageDescriptor{Syntax: proto2, FullName: fabupilot.antenna.PushVehicleStatusRequest, Fields: [{Name: status, Number: 1, Cardinality: optional, Kind: message, JSONName: "status", HasPresence: true, Message: fabupilot.antenna.VehicleStatus}]}
2024/08/05 20:14:19 [info] msg: 
2024/08/05 20:14:19 [info] pbMsg:
2024/08/05 20:14:19 [info] EncodedMessage:[10 215 9 10 11 104 111 119 111 49 49 95 116 101 115 116 18 36 9 0 0 0 0 0 0 0 0 17 0 0 0 0 0 0 0 0 25 67 146 203 96 197 194 24 65 33 255 242 59 5 119 36 73 65 25 0 0 0 0 0 0 0 0 32 2 42 13 99 97 109 101 114 97 95 100 114 105 118 101 114 42 11 103 110 115 115 95 100 114 105 118 101 114 42 12 108 105 100 97 114 95 100 114 105 118 101 114 42 7 118 101 104 105 99 108 101 48 0 56 187 165 215 178 249 130 181 244 23 65 52 210 89 163 229 42 3 192 80 2 97 0 0 0 0 0 0 89 64 104 1 138 1 138 2 10 38 9 119 21 230 217 96 151 217 65 18 14 97 103 101 110 116 95 100 121 110 97 109 105 99 115 24 246 137 19 73 178 21 230 217 96 151 217 65 57 0 0 0 0 0 0 0 0 65 0 0 0 0 0 0 0 0 81 0 0 0 0 0 0 0 0 89 0 0 0 0 0 0 0 0 153 1 0 0 0 0 0 0 0 0 161 1 52 210 89 163 229 42 3 192 185 1 67 146 203 96 197 194 24 65 193 1 255 242 59 5 119 36 73 65 245 1 0 0 0 0 253 1 0 0 0 0 133 2 0 0 0 0 128 3 1 234 3 126 9 160 241 87 63 197 194 24 65 17 50 99 95 1 119 36 73 65 25 0 0 0 0 0 0 0 0 33 25 139 248 18 183 91 33 192 41 0 0 0 0 0 0 0 0 49 0 0 0 0 0 0 0 0 57 0 0 0 0 0 0 0 0 65 0 0 0 0 0 0 0 0 73 0 0 0 0 0 0 0 0 81 0 0 0 0 0 0 0 0 89 0 0 0 0 0 0 0 0 97 0 0 0 0 0 0 0 0 105 0 0 0 0 0 0 0 0 113 110 199 69 4 140 249 26 191 146 1 35 8 1 122 10 18 8 16 192 132 61 24 192 132 61 138 1 18 9 140 77 229 217 96 151 217 65 73 140 77 229 217 96 151 217 65 154 1 6 8 0 16 0 24 0 176 1 2 186 1 129 1 10 18 9 189 25 235 217 96 151 217 65 73 189 25 235 217 96 151 217 65 25 150 117 124 32 22 13 22 192 41 0 0 0 0 0 0 0 0 49 0 0 0 0 0 0 0 0 57 0 0 0 0 0 0 0 0 65 0 0 0 0 0 0 0 0 73 0 0 0 0 0 0 0 0 81 0 0 0 0 0 0 0 0 104 0 112 1 120 2 145 1 0 0 0 0 0 0 0 0 169 1 0 0 0 0 0 0 89 64 216 1 0 145 2 0 0 0 0 0 0 0 0 162 2 4 96 0 104 0 200 1 1 210 1 75 82 27 10 17 80 79 82 84 95 77 69 73 83 72 65 78 95 68 79 67 75 18 4 100 111 99 107 24 5 90 9 76 65 78 69 95 75 69 69 80 98 16 78 79 82 77 65 76 95 76 65 78 69 95 75 69 69 80 106 15 80 82 69 67 73 83 69 45 80 65 82 75 73 78 71 216 1 5 232 1 1 242 1 36 9 0 0 0 0 0 0 0 0 17 0 0 0 0 0 0 0 0 25 67 146 203 96 197 194 24 65 33 255 242 59 5 119 36 73 65 250 1 51 10 23 112 111 114 116 95 109 101 105 115 104 97 110 95 100 114 105 118 101 114 108 101], pbMsg:
2024/08/05 20:14:19 [error] method:/fabupilot.antenna.AntennaServer/PushVehicleStatus, proto.Unmarshal:proto: cannot parse invalid wire-format data
2024/08/05 20:14:19 [error] get TCP pkg:127.0.0.1:37582 -> 127.0.0.1:6789, seq:2785354980, tcp flags:[PSH ACK]
1 3e05df44-5324-11ef-a5bb-04421a24b909 1722860059627710435
{"headers":{":authority":"127.0.0.1:6789",":method":"POST",":path":"/fabupilot.antenna.AntennaServer/PushVehicleStatus",":scheme":"http","content-type":"application/grpc","grpc-accept-encoding":"gzip","grpc-timeout":"3997656u","te":"trailers","user-agent":"grpc-java-netty/1.38.0"},"method":"/fabupilot.antenna.AntennaServer/PushVehicleStatus","request":"{}"}

{}
2024/08/05 20:14:24 [info] create Http2Conn, MaxDynamicTableSize:4096
2024/08/05 20:23:53 [info] create Http2Conn, MaxDynamicTableSize:4096
2024/08/05 20:23:53 [info] m: <nil>, ok: false
2024/08/05 20:23:53 [info] parseSymbol, svc:fabupilot.antenna.AntennaServer, method:PushHighFrameVehicleStatus
2024/08/05 20:23:53 [info] pfd: MessageDescriptor{Syntax: proto2, FullName: fabupilot.antenna.PushHighFrameVehicleStatusRequest, Fields: [{Name: high_frame_vehicle_status, Number: 1, Cardinality: optional, Kind: message, JSONName: "highFrameVehicleStatus", HasPresence: true, Message: fabupilot.antenna.HighFrameVehicleStatus}]}
2024/08/05 20:23:53 [info] pfmd: MessageDescriptor{Syntax: proto2, FullName: fabupilot.antenna.PushHighFrameVehicleStatusRequest, Fields: [{Name: high_frame_vehicle_status, Number: 1, Cardinality: optional, Kind: message, JSONName: "highFrameVehicleStatus", HasPresence: true, Message: fabupilot.antenna.HighFrameVehicleStatus}]}
2024/08/05 20:23:53 [info] msg: 
2024/08/05 20:23:53 [info] pbMsg:
2024/08/05 20:23:53 [info] EncodedMessage:[10 199 4 10 11 104 111 119 111 49 49 95 116 101 115 116 16 235 242 251 147 146 50 26 18 9 17 18 52 120 42 62 141 64 17 243 62 111 137 116 169 132 64 26 18 9 165 221 235 236 151 158 141 64 17 149 74 39 222 113 191 130 64 26 18 9 30 147 113 93 234 193 141 64 17 70 42 189 90 33 12 130 64 26 18 9 72 156 100 227 48 212 141 64 17 19 73 227 252 99 175 129 64 34 58 10 30 9 0 0 0 22 49 172 217 65 18 8 112 108 97 110 110 105 110 103 24 1 73 0 0 0 22 49 172 217 65 16 4 24 255 255 255 255 255 255 255 255 255 1 41 0 0 0 0 128 132 46 65 72 2 80 0 42 164 3 10 18 9 214 112 148 25 73 131 128 192 17 0 0 0 0 0 224 144 64 10 18 9 142 14 129 32 244 11 123 64 17 72 65 243 115 154 126 134 64 10 18 9 228 181 157 95 157 44 131 64 17 183 245 141 39 95 108 132 64 10 18 9 231 175 197 89 147 50 134 64 17 224 144 122 93 116 85 131 64 10 18 9 36 98 11 71 130 246 135 64 17 133 38 193 103 218 179 130 64 10 18 9 63 84 65 184 32 116 138 64 17 57 116 173 215 60 209 129 64 10 18 9 97 29 82 51 174 129 139 64 17 203 248 32 77 221 113 129 64 18 18 9 45 138 99 192 213 173 138 64 17 0 0 0 0 0 224 144 64 18 18 9 47 66 63 61 75 117 140 64 17 216 139 94 231 86 166 136 64 18 18 9 99 238 238 54 182 46 141 64 17 115 224 16 247 64 245 132 64 18 18 9 117 226 252 86 31 117 141 64 17 182 35 241 46 1 143 131 64 18 18 9 0 98 203 169 194 153 141 64 17 173 228 149 236 162 212 130 64 18 18 9 144 146 15 241 20 202 141 64 17 129 119 65 66 155 222 129 64 18 18 9 191 134 246 237 179 221 141 64 17 225 81 163 142 87 122 129 64 26 18 9 36 7 214 167 11 247 160 64 17 0 0 0 0 0 224 144 64 26 18 9 234 162 196 159 87 136 150 64 17 47 73 37 148 33 64 135 64 26 18 9 82 8 238 204 222 205 147 64 17 39 189 106 198 27 200 132 64 26 18 9 162 170 230 1 203 110 146 64 17 135 53 131 33 150 142 131 64 26 18 9 135 20 221 131 15 167 145 64 17 240 140 227 238 66 221 130 64 26 18 9 18 232 174 196 181 145 144 64 17 36 226 93 97 43 232 129 64 26 18 9 38 188 119 163 114 29 144 64 17 141 111 135 117 210 129 129 64], pbMsg:
1 93d15e78-5325-11ef-a5bb-04421a24b909 1722860633064204356
{"headers":{":authority":"127.0.0.1:6789",":method":"POST",":path":"/fabupilot.antenna.AntennaServer/PushHighFrameVehicleStatus",":scheme":"http","content-type":"application/grpc","grpc-accept-encoding":"gzip","grpc-timeout":"3997332u","te":"trailers","user-agent":"grpc-java-netty/1.38.0"},"method":"/fabupilot.antenna.AntennaServer/PushHighFrameVehicleStatus","request":"{\"highFrameVehicleStatus\":{\"vehicleName\":\"howo11_test\", \"timestampMs\":\"1722860632427\", \"planningTrajectoryOnCameraScreen\":[{\"x\":935.7707370822919, \"y\":661.1819027606092}, {\"x\":947.8241823603645, \"y\":599.9305995053898}, {\"x\":952.2394360421201, \"y\":577.516286352017}, {\"x\":954.5238712177461, \"y\":565.923822189014}], \"taskState\":{\"header\":{\"timestampSec\":1722860632, \"moduleName\":\"planning\", \"sequenceNum\":1, \"pubTimestampSec\":1722860632}, \"taskMode\":\"FAIL\", \"commandId\":-1, \"targetDist\":1000000, \"errorCode\":\"ROUTE_FAIL\", \"commandTimestamp\":\"0\"}, \"auxiliaryLane\":{\"leftPoint\":[{\"x\":-528.4106933209357, \"y\":1080}, {\"x\":432.7471013108524, \"y\":719.8254164699838}, {\"x\":613.576842529405, \"y\":653.5464621630298}, {\"x\":710.3219485706621, \"y\":618.6818189216101}, {\"x\":766.8136120690656, \"y\":598.4816432084923}, {\"x\":846.5159764388881, \"y\":570.154708247289}, {\"x\":880.2100588240829, \"y\":558.2330572677541}], \"midPoint\":[{\"x\":853.7293708588164, \"y\":1080}, {\"x\":910.6617379133294, \"y\":788.7924334894587}, {\"x\":933.8389719645023, \"y\":670.6567212408539}, {\"x\":942.6403026348695, \"y\":625.8755778159164}, {\"x\":947.220050419739, \"y\":602.5795528135692}, {\"x\":953.2602254120447, \"y\":571.8258099665255}, {\"x\":955.7128562221404, \"y\":559.2927525290835}], \"rightPoint\":[{\"x\":2171.522764862415, \"y\":1080}, {\"x\":1442.0855704044757, \"y\":744.0163958466491}, {\"x\":1267.4675786202029, \"y\":665.0135620440459}, {\"x\":1179.6982494393965, \"y\":625.8233061075888}, {\"x\":1129.7651514571933, \"y\":603.6576822068582}, {\"x\":1060.4275080994744, \"y\":573.0211818060529}, {\"x\":1031.3619517048442, \"y\":560.227763231365}]}}}"}

{}
2024/08/05 20:23:58 [info] create Http2Conn, MaxDynamicTableSize:4096


proto The file is not convenient to provide, can this information be analyzed

**Comment by fireblackman** [comment]
message PushVehicleStatusRequest {
  optional VehicleStatus status = 1;
}

message PushVehicleStatusResponse {
  optional ResponseBase base = 100;
}

message VehicleStatus {
  enum SensorStatus {
    OK = 1;
    ERROR = 2;
  }
  optional string vehicle_name = 1;
  optional Position position = 2;
  optional double speed = 3;
  optional SensorStatus sensor_status = 4;
  repeated string error_sensors = 5;
  optional bool system_ready = 6;
  optional int64 timestamp_ns = 7;
  optional double heading = 8;
  repeated SensorInfo sensor_info = 9;
  enum DrivingStatus {
    OFF_LINE = 0;
    MANUAL = 1;
    AUTO_DRIVE = 2;
    REMOTE_CTRONL = 3;
    EMERGENCY_STOP = 4;
  }
  optional DrivingStatus driving_status = 10;
  repeated fabupilot.common.PointENU fit_route_points = 11;
  optional double soc = 12;
  optional bool no_load = 13;
  enum VehicleOwner {
    FABU_AUTO_PILIOT = 0;
    PORT_MANUAL = 1;
  }
  optional VehicleOwner vehicle_owner = 14 [default = FABU_AUTO_PILIOT];
  optional double total_mileage = 15;
  optional double current_mileage = 16;
  optional fabupilot.localization.Localization localization_detail = 17;
  optional fabupilot.canbus.ChassisDetail chassis_detail = 18;
  message ContainerExistStatus {
    optional bool front_exist = 1 [default = false];
    optional bool mid_exist = 2 [default = false];
    optional bool rear_exist = 3 [default = false];
  }
  optional ContainerExistStatus container_status = 19;
  repeated string node_path = 20;  // for ali iecs
  repeated Position path1 = 21;    // for ali iecs
  optional int32 net_delay = 22;   // ms
  optional fabupilot.canbus.Chassis chassis = 23;
  message TargetDistance {
    optional double target_dist = 1 [default = 1e+6];
    optional double away_dist = 2 [default = 0];
  }
  optional TargetDistance target_distance = 24;
  optional bool driverless = 25;
  optional BehaviorInfoSub behavior_info_sub = 26; // receive from PlanningInfo
  enum WorkMode {
    NORMAL_WORK = 0; //正常工作模式
    ABNORMAL_WORK = 1;//异常工作模式
    STOP_WORK = 2; //收工模式
    LEAVE_WORK = 3; //离场模式
    DORMANCY_WORK = 4;//休眠模式
    OFFLINE_WORK = 5;
  }
  optional WorkMode work_mode = 27 [default = NORMAL_WORK];
  optional int64 receive_timestamp_ns = 28; // antenna server receive time
  optional bool can_enter_auto_drive_mode = 29;
  optional string error_info_for_enter_auto_drive_mode = 41;

  // 挂车前侧中心点，南通用于计算车辆所属TP
  optional Position trailer_front_center = 30;
  
  message VehicleMode {
    optional string business_scene = 1;       //场景
    optional bool safe_mode = 2;              // 安全模式
    optional bool dynamic_vessel_status = 3;  // 动态船头船尾
    optional int32 bridge_driving_mode =
        4;  // 潮汐车道，之前只有11号引桥，后续扩充10号引桥，该字段默认填11号引桥模式，云端适配后废弃该字段
    optional bool air_pump_status = 5;          // 气动开关
    repeated BridgeDrivingMode bridges_driving_mode = 6;
    optional bool use_online_routing = 7;              // 新云控模式
  }
  optional VehicleMode vehicle_mode = 31;
  
  message DeviceInfo {
    message HostInfo {
      optional string ip = 1;//ip
      optional string version = 2;//软件版本信息
      optional double version_timestamp_sec = 3;//版本时间秒
    }
    repeated HostInfo host_info = 1; //主机信息
    enum EnvMode {
      PRO = 0;//生产
      PRE = 1;//预发
      FAT = 2;//测试
      DEV = 3;//开发
    }
    optional EnvMode env_mode = 2; //部署环境
  }
  optional DeviceInfo device_info = 32; //设备信息

  optional LightColor light_color = 33; // 车辆警示灯
  // 该指令类型专用于产品区分车辆作业类型
  enum CommandType {
    UNKNOWN = 0;
    REMOTE_CONTROL = 1;  // 当前仅打点指令在产品层面归属远程控制状态
  }
  optional CommandType command_type = 34;
  optional double trailer_heading = 35;
  optional int32 fleet_id = 36; // 所属车队
  optional fabupilot.antenna.TaskState task_state = 37;
  message TrajectoryPoint {
    optional double timestamp = 1;
    optional double pos_x = 2;
    optional double pos_y = 3;
    optional double heading = 4;
    optional double trailer_angle = 5;
    optional double vel = 6;
  }
  repeated TrajectoryPoint candidate_route_start_point = 38;
  repeated TrajectoryPoint trajectory_points = 39;
  optional DistToPlcDevice dist_to_plc_device = 40;
  optional int64 startup_timestamp_ns = 42; //antenna启动时间
  optional fabupilot.common.VehicleInfo.VehicleType vehicle_type = 43; // 车辆类型，比如集卡、IGV等
}

**Comment by vearne** [comment]
I will try to analyze the cause of the problem

**Comment by vearne** [comment]
vehicle_status.proto:80:10: "BridgeDrivingMode" is not defined.
vehicle_status.proto:12:10: "Position" is not defined.
vehicle_status.proto:19:10: "SensorInfo" is not defined.
vehicle_status.proto:28:10: "fabupilot.common.PointENU" is not defined.
vehicle_status.proto:38:10: "fabupilot.localization.Localization" is not defined.
vehicle_status.proto:39:10: "fabupilot.canbus.ChassisDetail" is not defined.
vehicle_status.proto:47:10: "Position" is not defined.
vehicle_status.proto:49:10: "fabupilot.canbus.Chassis" is not defined.
vehicle_status.proto:56:10: "BehaviorInfoSub" is not defined.
vehicle_status.proto:71:10: "Position" is not defined.
vehicle_status.proto:102:10: "LightColor" is not defined.
vehicle_status.proto:111:10: "fabupilot.antenna.TaskState" is not defined.
vehicle_status.proto:122:10: "DistToPlcDevice" is not defined.
vehicle_status.proto:124:10: "fabupilot.common.VehicleInfo.VehicleType" is not defined.

Some definitions seem to be missing

**Comment by fireblackman** [comment]
This is a complex structure. I will separate the interface to provide you

**Comment by vearne** [comment]
> 10 199 4 10 11 104 111 119 111 49 49 95 116 101 115 116 16 235 242 251 147 146 50 26 18 9 17 18 52 120 42 62 141 64 17 243 62 111 137 116 169 132 64 26 18 9 165 221 235 236 151 158 141 64 17 149 74 39 222 113 191 130 64 26 18 9 30 147 113 93 234 193 141 64 17 70 42 189 90 33 12 130 64 26 18 9 72 156 100 227 48 212 141 64 17 19 73 227 252 99 175 129 64 34 58 10 30 9 0 0 0 22 49 172 217 65 18 8 112 108 97 110 110 105 110 103 24 1 73 0 0 0 22 49 172 217 65 16 4 24 255 255 255 255 255 255 255 255 255 1 41 0 0 0 0 128 132 46 65 72 2 80 0 42 164 3 10 18 9 214 112 148 25 73 131 128 192 17 0 0 0 0 0 224 144 64 10 18 9 142 14 129 32 244 11 123 64 17 72 65 243 115 154 126 134 64 10 18 9 228 181 157 95 157 44 131 64 17 183 245 141 39 95 108 132 64 10 18 9 231 175 197 89 147 50 134 64 17 224 144 122 93 116 85 131 64 10 18 9 36 98 11 71 130 246 135 64 17 133 38 193 103 218 179 130 64 10 18 9 63 84 65 184 32 116 138 64 17 57 116 173 215 60 209 129 64 10 18 9 97 29 82 51 174 129 139 64 17 203 248 32 77 221 113 129 64 18 18 9 45 138 99 192 213 173 138 64 17 0 0 0 0 0 224 144 64 18 18 9 47 66 63 61 75 117 140 64 17 216 139 94 231 86 166 136 64 18 18 9 99 238 238 54 182 46 141 64 17 115 224 16 247 64 245 132 64 18 18 9 117 226 252 86 31 117 141 64 17 182 35 241 46 1 143 131 64 18 18 9 0 98 203 169 194 153 141 64 17 173 228 149 236 162 212 130 64 18 18 9 144 146 15 241 20 202 141 64 17 129 119 65 66 155 222 129 64 18 18 9 191 134 246 237 179 221 141 64 17 225 81 163 142 87 122 129 64 26 18 9 36 7 214 167 11 247 160 64 17 0 0 0 0 0 224 144 64 26 18 9 234 162 196 159 87 136 150 64 17 47 73 37 148 33 64 135 64 26 18 9 82 8 238 204 222 205 147 64 17 39 189 106 198 27 200 132 64 26 18 9 162 170 230 1 203 110 146 64 17 135 53 131 33 150 142 131 64 26 18 9 135 20 221 131 15 167 145 64 17 240 140 227 238 66 221 130 64 26 18 9 18 232 174 196 181 145 144 64 17 36 226 93 97 43 232 129 64 26 18 9 38 188 119 163 114 29 144 64 17 141 111 135 117 210 129 129 64

```
protoc --decode_raw < vehicle_status.bin
```
vehicle_status.bin is the serialized data file
The protoc --decode_raw command can decode the raw Protobuf data and output it as a digitally labeled message. This method does not rely on the .proto file, so it can help you view all fields and values, but you may not be able to recognize the actual meaning of the field.

Deserialization failed, indicating that there is indeed a problem with the message format

I have two possibilities
1) /xxxxxx.antenna.AntennaServer/PushVehicleStatus is a streaming RPC, not a Unary RPC, vearne/grpcreplay currently only supports Unary RPC
2) The request body is too large, it is split into multiple TCP packets

**Comment by fireblackman** [comment]
rpc_method_handlers = {
            'PushVehicleStatus': grpc.unary_unary_rpc_method_handler(
                    servicer.PushVehicleStatus,
                    request_deserializer=antenna__pb2.PushVehicleStatusRequest.FromString,
                    response_serializer=antenna__pb2.PushVehicleStatusResponse.SerializeToString,
            )

may be 2 is reason
The proto files are interdependent and separation takes some time

**Comment by vearne** [comment]
fix bug that request body is too big, use v0.1.2 and try again @fireblackman

**Comment by fireblackman** [comment]
it seems not work.

**Comment by fireblackman** [comment]
gmail Returned my email, I put it here, please verify it
[antenna-server1.zip](https://github.com/user-attachments/files/16510152/antenna-server1.zip)

**Comment by vearne** [comment]
> gmail Returned my email, I put it here, please verify it [antenna-server1.zip](https://github.com/user-attachments/files/16510152/antenna-server1.zip)

give me a client script to access server @fireblackman

**Comment by vearne** [comment]
1 83ca6f5a-5470-11ef-ad91-5626e1cdcfe2 1723002769592714000
{"headers":{":authority":"127.0.0.1:8888",":method":"POST",":path":"/fabupilot.antenna.AntennaServer/PushVehicleStatus",":scheme":"http","content-type":"application/grpc","grpc-accept-encoding":"identity, deflate, gzip","te":"trailers","user-agent":"grpc-python/1.65.4 grpc-c/42.0.0 (osx; chttp2)"},"method":"/fabupilot.antenna.AntennaServer/PushVehicleStatus","request":"{\"status\":{\"vehicleName\":\"xxxx\"}}"}

**Comment by fireblackman** [comment]
https://github.com/bloomrpc/bloomrpc/releases/tag/1.5.3
i use bloomrpc to  request is ok. but use jmeter is failed.
[grpc接口调试脚本.zip](https://github.com/user-attachments/files/16519579/grpc.zip)
jmeter-5.6.3
need grpc-request plugin
![image](https://github.com/user-attachments/assets/4b2efd7f-1ba0-478f-8f5b-364ff0e9643f)

**Comment by fireblackman** [comment]
2024/08/07 14:57:04 [info] create Http2Conn, MaxDynamicTableSize:4096
1 413e8cc2-548a-11ef-9d2e-04421a24b909 1723013824861306201
{"headers":{":authority":"127.0.0.1:8888",":method":"POST",":path":"/fabupilot.antenna.AntennaServer/PushVehicleStatus",":scheme":"http","accept-encoding":"identity,gzip","content-type":"application/grpc","grpc-accept-encoding":"identity,deflate,gzip","te":"trailers","user-agent":"grpc-node/1.24.7 grpc-c/8.0.0 (linux; chttp2; ganges)"},"method":"/fabupilot.antenna.AntennaServer/PushVehicleStatus","request":"{\"status\":{\"position\":{\"longitude\":0, \"latitude\":0}, \"speed\":0, \"heading\":-2.3959458123956896, \"soc\":100, \"chassis\":{\"header\":{\"version\":1}, \"soc\":100, \"rainlevel\":\"NORAIN\", \"deceleration\":0, \"signal\":{\"horn\":false}}, \"driverless\":true}}"}

Does the output of the body do the length limit？

**Comment by fireblackman** [comment]
use grpcurl also happened.
[antenna_client.zip](https://github.com/user-attachments/files/16520152/antenna_client.zip)
maybe you do not to  download  jmeter.

**Comment by vearne** [comment]
The bug has been fixed. use v0.1.3 and try again @fireblackman 

grpcreplay depends on github.com/google/gopacket
When gopacket specifies a network card to capture packets, there is a parameter snapshotLen that specifies the maximum number of bytes to read for a single packet. If the limit is exceeded, the data will be discarded.

snapshotLen was originally set to 1024, but I have now changed it to 20000
```
func (l *DeviceListener) listen() error {
	var err error
	l.handle, err = pcap.OpenLive(l.device, snapshotLen, promiscuous, timeout)
	if err != nil {
		return err
	}
        ...
}
```
